scene/resources/motion_blur_compositor_effect.h

```diff
@@ -0,0 +1,77 @@
+/*  motion_blur_compositor_effect.h                                       */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef MOTION_BLUR_COMPOSITOR_EFFECT_H
+#define MOTION_BLUR_COMPOSITOR_EFFECT_H
+
+#include "core/os/mutex.h"
+#include "scene/resources/compositor.h"
+#include "servers/rendering/renderer_rd/storage_rd/render_buffer_custom_data_rd.h"
+#include "servers/rendering/renderer_rd/storage_rd/render_data_rd.h"
+#include "servers/rendering/rendering_device.h"
+#include "servers/rendering/storage/render_scene_data.h"
+
+class MotionBlurCompositorEffect : public CompositorEffect {
+	GDCLASS(MotionBlurCompositorEffect, CompositorEffect);
+
+private:
+	RID linear_sampler;
+
+	RID motion_blur_shader;
+	RID motion_blur_pipeline;
+
+	RID overlay_shader;
+	RID overlay_pipeline;
+
+	StringName context = "MotionBlur";
+	StringName texture = "texture";
```

Comment:
These should be in the cpp and as `SNAME`s in the method they're used in IMO, they're only used in one

Reply:
I'm not sure what the current recommendation is so I make them const values that are strings.